tests/ha/single_pod_restart/test_server_pod_restart.py

```diff
@@ -983,3 +983,476 @@ def test_writes_during_server_pod_restart(self):
+        buckets = s3_test_obj.bucket_list()[1]
+        assert_utils.assert_equal(len(buckets), wr_bucket, f"Failed to create {wr_bucket} number "
+                                                           f"of buckets. Created {len(buckets)} "
+                                                           f"number of buckets")
+        LOGGER.info("Step 3: Successfully performed WRITEs with variable object sizes.")
+        LOGGER.info("Step 4: Start Continuous DELETEs in background on %s buckets", del_bucket)
+        get_random_buck = self.system_random.sample(buckets, del_bucket)
+        args = {'test_prefix': self.test_prefix_deg, 'test_dir_path': self.test_dir_path,
+                'bkt_list': get_random_buck, 'skipput': True, 'skipget': True,
+                'bkts_to_del': del_bucket, 'output': del_output}
+        thread = threading.Thread(target=self.ha_obj.put_get_delete,
+                                  args=(event, s3_test_obj,), kwargs=args)
+        thread.daemon = True  # Daemonize thread
+        thread.start()
+        LOGGER.info("Step 4: Successfully started continuous DELETEs in background on %s buckets",
+                    del_bucket)
+        LOGGER.info("Step 5: Restart the pod with replica method and check cluster status")
+        event.set()
+        resp = self.ha_obj.restore_pod(pod_obj=self.node_master_list[0],
+                                       restore_method=self.restore_method,
+                                       restore_params={"deployment_name": self.deployment_name,
+                                                       "deployment_backup":
+                                                           self.deployment_backup,
+                                                       "num_replica": self.num_replica,
+                                                       "set_name": self.set_name},
+                                       clstr_status=True)
+        LOGGER.debug("Response: %s", resp)
+        assert_utils.assert_true(resp[0], f"Failed to restore pod by {self.restore_method} way")
+        self.restore_pod = False
+        LOGGER.info("Step 5: Successfully restart the pod with replica method and checked "
+                    "cluster status")
+        event.clear()
+        thread.join()
+        LOGGER.info("Step 6: Verify status for In-flight DELETEs while pod was restarting are "
+                    "successful & check the remaining buckets.")
+        del_resp = tuple()
+        while len(del_resp) != 2:
+            del_resp = del_output.get(timeout=HA_CFG["common_params"]["60sec_delay"])
+        if not del_resp:
+            assert_utils.assert_true(False, "Background process failed")
+        event_del_bkt = del_resp[0]
+        fail_del_bkt = del_resp[1]
+        assert_utils.assert_false(len(fail_del_bkt) or len(event_del_bkt),
+                                  "Bucket deletion failed while server pod restart"
```

Comment:
not only while server pod restart right? even before after as we are checking for both event and fail??? so either we can seperate it or just say failed in server pod restart process